refactor: Replace react-bootstrap tabs with Antd tabs on Profile

kgabryje opened the pull request:
### SUMMARY
This PR re-reverts part of PR https://github.com/apache/incubator-superset/pull/11090 - it contains only changes to the Profile view, which were not affected by the issue https://github.com/apache/incubator-superset/issues/11192

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before/after:

![image](https://user-images.githubusercontent.com/15073128/95556359-7e9ff280-0a13-11eb-8a97-523c72986d91.png)
![image](https://user-images.githubusercontent.com/15073128/95556388-852e6a00-0a13-11eb-8420-5b84343ba8eb.png)

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

kgabryje committed:
Create styled Antd Tabs and Dropdown

kgabryje committed:
Replace Tabs in Profile view

rusackas commented on superset-frontend/src/common/components/Dropdown.tsx:
```diff
@@ -0,0 +1,78 @@
+  height: 3px;
+  border-radius: 1.5px;
+  background-color: #bababa;
+`;
```
Would like to see these four properties pulled from the theme... i.e. gridUnit (4px) and `grayscale.light1` for the gray. Also, fractional pixels round to the nearest pixel in browsers at 100%. Unless we're trying to get fancy with zooming, 2px  (or gridUnit / 2) should work here.

kgabryje replied:
Done. I had to move these styles from `css` to `styled`, because it seems that I can't use theme variables in `css` function 🙁

rusackas commented on superset-frontend/src/common/components/Modal.tsx:
```diff
@@ -81,6 +81,11 @@ const StyledModal = styled(BaseModal)`
       margin-left: 8px;
```
gridUnit *2, while you're in there :)

kgabryje replied:
Done here and in the whole Modal file

rusackas commented on superset-frontend/src/common/components/Tabs.tsx:
```diff
@@ -16,27 +16,46 @@
+import React from 'react';
 import { styled } from '@superset-ui/core';
 import { Tabs as AntdTabs } from 'src/common/components';
+import { css } from '@emotion/core';
```
You can import this from @superset-ui/core along with styled a couple lines up.

kgabryje replied:
Done

rusackas commented on superset-frontend/src/common/components/Modal.tsx:
```diff
@@ -81,6 +81,11 @@ const StyledModal = styled(BaseModal)`
+
+  // styling for Tabs component
+  .ant-tabs {
+    margin-top: -18px;
```
Not sure the context of this exact measurement, or if gridUnit makes sense to use here too.

kgabryje replied:
I changed it to `-gridUnit*4`, I guess 2px won't make a big difference

rusackas commented on superset-frontend/src/common/components/Tabs.tsx:
```diff
@@ -16,27 +16,46 @@
 import { styled } from '@superset-ui/core';
 import { Tabs as AntdTabs } from 'src/common/components';
+import { css } from '@emotion/core';
+import Icon from '../../components/Icon';
```
```suggestion
import Icon from 'src/components/Icon';
```
Using this webpack alias makes it easier to shuffle files around :)

kgabryje replied:
Done

rusackas reviewed (changes requested):
Just a few suggestions/nits to review, but generally looking good.

kgabryje committed:
Change some hard coded variables to theme

kgabryje commented:
@rusackas I did the changes that you requested. Frankly, I'm not sure why some of these files (like Modal and Dropdown) were a part of this PR - they were gone after I rebased to the newest master. I changed these hard-coded variables nevertheless 🙂

kgabryje committed:
Lint fix

rusackas reviewed (approved):
There are a few remaining CSS attributes in this code that I wish were theme variables (font-sizes, in particular), but it looks like all the NEW styles are covered, and the hard coded colors are gone so I'm happy enough! LGTM, onward and upward!
